src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -100,7 +100,11 @@ public void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Ex
         try {
             validateEmbeddingFieldsValue(ingestDocument);
             Map<String, Object> knnMap = buildMapWithKnnKeyAndOriginalValue(ingestDocument);
-            mlCommonsClientAccessor.inferenceSentences(this.modelId, createInferenceList(knnMap), ActionListener.wrap(vectors -> {
+            List<String> inferenceList = createInferenceList(knnMap);
```

Comment:
Can we validate and see what will happen if knnMap is empty or null?

Reply:
The knnMap won't be null since it's manually created, but it could be empty if user doesn't configure anything in the pipeline field_map. This case the inference list will be empty.